pkg/storage/snapshot/snapshot_test.go

```diff
@@ -2307,6 +2308,60 @@ var _ = Describe("Snapshot controlleer", func() {
 	})
 })
 
+func applyPatch(patch []byte, orig, patched interface{}) error {
+	originalBytes, err := json.Marshal(orig)
+	if err != nil {
+		return err
+	}
+	patchJSON, err := jsonpatch.DecodePatch(patch)
+	if err != nil {
+		return err
+	}
+	patchedBytes, err := patchJSON.Apply(originalBytes)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(GinkgoWriter, "Patched object: %s\n", string(patchedBytes))
+
+	if err = json.Unmarshal(patchedBytes, patched); err != nil {
+		return err
+	}
+	return nil
+}
+
+func expectVMSnapshotPatch(client *kubevirtfake.Clientset, orig, newObj *snapshotv1.VirtualMachineSnapshot) {
+	client.Fake.PrependReactor("patch", "virtualmachinesnapshots", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+		patch, ok := action.(testing.PatchAction)
+		Expect(ok).To(BeTrue())
+
+		patched := &snapshotv1.VirtualMachineSnapshot{}
+		if err := applyPatch(patch.GetPatch(), orig, patched); err != nil {
+			return false, nil, err
+		}
+
+		Expect(patched).To(Equal(newObj))
+
+		return reflect.DeepEqual(newObj, patched), patched, nil
+	})
+}
+
+func expectVMSnapshotContentPatch(client *kubevirtfake.Clientset, orig, newObj *snapshotv1.VirtualMachineSnapshotContent) {
+	client.Fake.PrependReactor("patch", "virtualmachinesnapshotcontents", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+		patch, ok := action.(testing.PatchAction)
+		Expect(ok).To(BeTrue())
+
+		patched := &snapshotv1.VirtualMachineSnapshotContent{}
+		if err := applyPatch(patch.GetPatch(), orig, patched); err != nil {
+			return false, nil, err
+		}
+
+		Expect(patched).To(Equal(newObj))
+
+		return reflect.DeepEqual(newObj, patched), patched, nil
+	})
+}
+
 func expectVMSnapshotUpdate(client *kubevirtfake.Clientset, vmSnapshot *snapshotv1.VirtualMachineSnapshot) {
```

Comment:
seems like we should not expect update on virtualmachinesnapshot? can we delete this? I see it is still used in some unit tests though you removed all the updates?

Reply:
The unit test reactor framework does not differentiate between update and updateStatus calls. They both map to `UpdateAction`. So this function is still very much needed

Reply:
this may help https://github.com/kubevirt/kubevirt/blob/86e267d37c06df0eecdf6c8aed5746268efd66da/pkg/virt-controller/watch/workload-updater/workload-updater_test.go#L56-L57

Reply:
Given how the tests are written, it shouldn't matter much but put explicit subresource check in there